pkg/diagnostics/tracing.go

```diff
@@ -31,6 +31,9 @@ const (
 	// CorrelationID is the header key name of correlation id for trace
 	CorrelationID      = "X-Correlation-ID"
 	correlationKey key = CorrelationID
+	// CtxCorrelationIDKey is the key for correlationId in the request context
+	CtxCorrelationID        = "CtxCorrelationID"
```

Comment:
What is the relationship between CorrelationID & CtxCorrelationID and why these are separate?

Reply:
+1 This is not a standard way. The way to get correlation id is different for http and gRPC. I would create the helper to get correlation id from the context id per protocol later instead of introducing our own thing.